### ISSUE TITLE
Customizable <select>'s button and selectedcontent are not treated as inert, causing false button-name violations

### ISSUE THREAD

**Issue by ItsMeSousuke** [dequelabs/axe-core#5315]
### Product

axe-core

### Product Version

4.13.0

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation

`axe.run()` should not report a `button-name` violation for the internal `<button>` used by a customizable `<select>` (`<select><button>…</button><option>…</option></select>`). Per the HTML spec, that `<button>` and its `<selectedcontent>` child are inert — excluded from the accessibility tree and not focusable — so `button-name` (and any other rule relying on accessibility-tree visibility or focusability) should treat them as inapplicable, not as a naming violation.

#### Actual

axe-core has no awareness of this select-specific inert rule. `lib/commons/dom/is-inert.js` only accounts for the `inert` attribute and elements outside an open modal `<dialog>`. As a result, the scan sees a `<button>` with no text, no `aria-label`, and no wrapping `<label>`, and reports a `critical` `button-name` violation on it. The same gap can also affect any other rule that depends on `isVisibleToScreenReaders`, `isFocusable`, or `isInert` for this subtree (e.g. `nested-interactive`, `color-contrast`).

#### How to Reproduce

```js
axe.run(document.body).then(results => {
  console.log(results.violations);
});
```

against this fixture:

```html
<select>
  <button><selectedcontent></selectedcontent></button>
  <option>Apple</option>
  <option>Banana</option>
</select>
```

`results.violations` includes a `button-name` violation for the `<button>`, even though it is spec-inert and never exposed to assistive technology.

#### Additional context

Suggested fix: teach `isInert` (`lib/commons/dom/is-inert.js`) that any `<button>` that is a direct child of a `<select>`, and any `<selectedcontent>` element, are inert per [the HTML spec's `select` element definition](https://html.spec.whatwg.org/multipage/form-elements.html#the-select-element). This should apply regardless of the select's rendered `appearance` (`base-select` vs. the default UA widget) — inertness comes from being a `select`'s button/selectedcontent, not from how it's styled. This propagates automatically to `isVisibleToScreenReaders` and `focusDisabled`/`isFocusable`, which already delegate to `isInert`. `axe.runVirtualRule('button-name', …)` bypasses the gather-time visibility filter, so `button-name`'s `matches` function should also be updated to skip inert nodes directly.

Related: #4836 tracks a separate, complementary ARIA-in-HTML standards update (disallowing ARIA roles/attributes on `selectedcontent`/the select button, plus a `role=generic` fallback for a misused `selectedcontent`) — that issue is about `standards/` data and does not cover this inert/focusability gap.
